modules/simphera_aws_instance/postgresql.tf

```diff
@@ -1,3 +1,17 @@
+data "aws_subnet" "vpc_private_subnet" {
+  for_each = toset(var.private_subnets)
+  id       = each.value
+}
+
+resource "aws_db_subnet_group" "database" {
+  name       = "${var.infrastructurename}-vpc"
+  subnet_ids = [for s in data.aws_subnet.vpc_private_subnet : s.id]

```

Comment:
Can you use `module.vpc.private_subnets`? In that case you would not need `data.aws_subnet`

Reply:
Yes, a good catch, it will not work with clean deployment now, when filter is replaced with var. Let me fix that.

Reply:
Fixed